gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -865,4 +1016,23 @@ public static Builder builder() {
    * @throws StorageException upon failure
    */
   List<Boolean> delete(BlobId... blobIds);
+
+  /**
+   * Returns a {@link BlobRewriter} object for the provided {@code rewriteRequest}. If source and
+   * destination objects share the same location and storage class the source blob is copied with a
+   * single call of {@link BlobRewriter#copyChunk()}, regardless of the {@link
+   * RewriteRequest#maxBytesRewrittenPerCall} parameter. If source and destination have different
+   * location or storage class multiple RPC calls might be needed depending on blob's size.
+   * <p>
+   * Example usage of blob rewriter:
+   * <pre>    {@code BlobRewriter rewriter = service.rewriter(rewriteRequest);}
+   *    {@code while(!rewriter.isDone()) {
+   *       rewriter.copyChunk();
+   *   }}
+   * </pre>
+   *
+   * @throws StorageException upon failure
+   * @see <a href="https://cloud.google.com/storage/docs/json_api/v1/objects/rewrite">Rewrite</a>
+   */
+  BlobRewriter rewriter(RewriteRequest rewriteRequest);
```

Comment:
I'd much prefer you call this low-level wrapping of Rewrite the same name as the API method it's replacing. If you want to synthesize a higher-level Copy that uses Rewrite under the covers but doesn't expose the rewrite token to the client, that seems entirely reasonable. Since under the covers, Copy uses Rewrite and simply fails if Rewrite cannot complete in one request, uniformly using Rewrite seems like the right thing to do until such time as our service offers a service-side alternative.